src/sse42_deser.rs

```diff
@@ -0,0 +1,209 @@
+#![deny(warnings)]
+
+#[cfg(target_arch = "x86")]
+use std::arch::x86::*;
+#[cfg(target_arch = "x86_64")]
+use std::arch::x86_64::*;
+
+use std::mem;
+
+pub use crate::error::{Error, ErrorType};
+pub use crate::Deserializer;
+pub use crate::Result;
+pub use crate::sse42_utf8check::*;
+pub use crate::stringparse::*;
+
+use crate::portability::trailingzeroes;
+
+
+impl<'de> Deserializer<'de> {
+    #[cfg_attr(not(feature = "no-inline"), inline(always))]
+    pub fn parse_str_(&mut self) -> Result<&'de str> {
```

Comment:
parse_str_ is the only method currently platform-specific in Deserializer (made it public to ease testing, might not have to be that way)